src/compiler/crystal/tools/doc/generator.cr

```diff
@@ -241,6 +251,17 @@ class Crystal::Doc::Generator
     end
   end
 
+  # adds extra line break to TODO lines
+  def isolate_flag_lines(string)
+    string.split(/\n/).map do |line|
```

Comment:
I feel like the regex here and the regex on line 235-240 could be replaced by a method which takes a line and returns `{flag, text}` or `nil`. We can even combine them, as HTML passes through markdown unimpeded.

For example:
```cr
if flag, text = parse_flag(line)
  io << %(<span class="flag flag-#{flag.downcase}">#{flag}</span> #{text}\n)
else
  io << line
end
```

which requires only one pass.